client/my-sites/checkout/checkout-thank-you/pending.jsx

```diff
@@ -0,0 +1,109 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import page from 'page';
+import { localize } from 'i18n-calypso';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { identity } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getOrderTransaction, getOrderTransactionError } from 'state/selectors';
+import { ORDER_TRANSACTION_STATUS } from 'state/order-transactions/constants';
+import { errorNotice } from 'state/notices/actions';
+
+class CheckoutPending extends PureComponent {
+	static propTypes = {
+		orderId: PropTypes.number.isRequired,
+		siteSlug: PropTypes.string.isRequired,
+		transaction: PropTypes.object,
+		error: PropTypes.object,
+		errorNotice: PropTypes.func,
+		localize: PropTypes.func,
+	};
+
+	static defaultProps = {
+		localize: identity,
+		errorNotice: identity,
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		const { transaction, error } = nextProps;
+		const { translate, showErrorNotice } = this.props;
+
+		const retryOnError = () => {
+			page( `/checkout/${ this.props.siteSlug }` );
+
+			showErrorNotice(
+				translate( "Sorry, we couldn't process your payment. Please try again later." )
+			);
+		};
+
+		if ( transaction ) {
+			const { processingStatus } = transaction;
+
+			if ( ORDER_TRANSACTION_STATUS.SUCCESS === processingStatus ) {
+				page( `/checkout/thank-you/${ this.props.siteSlug }` );
+				return;
+			}
+
+			// It is mostly because the user has cancelled the payment.
+			// See the explanation in https://github.com/Automattic/wp-calypso/pull/23670#issuecomment-377186515
+			if ( ORDER_TRANSACTION_STATUS.FAILURE === processingStatus ) {
+				// Bring the user back to the homepage in this case.
+				page( '/' );
```

Comment:
I'm having second doubts about this - maybe the plans page is better?

Reply:
After playing a bit more, I agree with you. Since the whole flow is initiated from the plan page, redirecting back there is much less confusing. I've updated it in 9fc71cf. The case of unknown status has been updated as well, which I think the same reasoning applied :)